iothub/device/src/HsmAuthentication/HttpHsmComunicationException.cs

```diff
@@ -20,6 +22,16 @@ public HttpHsmComunicationException(string message, int statusCode) : base($"{me
             StatusCode = statusCode;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpHsmComunicationException"/> class with the specified serialization and context information.
+        /// </summary>
+        /// <param name="info">An object that holds the serialized object data about the exception being thrown.</param>
+        /// <param name="context">An object that contains contextual information about the source or destination.</param>
+        protected HttpHsmComunicationException(SerializationInfo info, StreamingContext context)
```

Comment:
Since the exception type is not a sealed class, the serialization info constructor needs to be `protected`.